index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
+      return parseFloat(num.toFixed(n)).toString();
+    }
+  }
+
+  function equation() {
+    if (currentOp == "add") {
+      currentString = firstNum + Number(currentString);
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+    } else if (currentOp == "subtract") {
+      currentString = firstNum - Number(currentString);
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+    } else if (currentOp == "multiply") {
+      currentString = firstNum * Number(currentString);
+      numDisplay.textContent = rounder(Number(currentString));
+      firstNum = Number(currentString);
+      currentOp = "";
+      currentString = "";
+    } else if (currentOp == "divide") {
+      if (Number(currentString) == 0) {
+        numDisplay.textContent = "Error";
+        firstNum = "";
+        currentOp = "";
+        currentString = "";
+        errLock = true;
+      } else {
+        currentString = firstNum / Number(currentString);
+        numDisplay.textContent = rounder(Number(currentString));
+        firstNum = Number(currentString);
+        currentOp = "";
+        currentString = "";
+      }
+    } else {
+      return;
+    }
+  }
 
   for (let n = 0; n <= 9; n++) {
     numButtons[n].addEventListener("click", () => {
-      if (currentString.length <= 10) {
-        if (errLock == true) {
-          currentString = "";
-          errLock = false;
-          currentString = currentString + numButtons[n].textContent;
-          numDisplay.textContent = currentString;
-        } else {
-          currentString = currentString + numButtons[n].textContent;
-          numDisplay.textContent = currentString;
-        }
+      if (errLock == true) {
+        errLock = false;
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString == "") {
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString == "" || currentString == "0") {
+        currentString = numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
+      } else if (currentString.match(/\d/g).length < 10) {
+        //If the currentString contains less than 10 digits (excluding decimal point and negative sign)
+        currentString = currentString + numButtons[n].textContent;
+        numDisplay.textContent = currentString;
+        clearCheck();
       } else {
-        return;
+        currentString = currentString + numButtons[n].textContent;
+        clearCheck();
+        //Else, we will continue to add digits to the currentString, but NOT to the actual display (out of room)
```

Comment:
I don't think you should do this. It would be confusing to the user if their actual typed number differed from what was displayed. If the display is full, don't update currentString either.

Reply:
Okay. Updated the else to just return.